packages/react-dom/src/fire/ReactFireEventTraversal.js

```diff
@@ -0,0 +1,234 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import {
+  getClosestFiberFromDOMNode,
+  getDOMNodeFromFiber,
+} from './ReactFireInternal';
+import {getDomNodeEventsMap} from './ReactFireEvents';
+
+import {HostComponent, HostRoot} from 'shared/ReactWorkTags';
+import {invokeGuardedCallbackAndCatchFirstError} from 'shared/ReactErrorUtils';
+
+function getParent(fiber) {
+  do {
+    fiber = fiber.return;
+    // TODO: If this is a HostRoot we might want to bail out.
+    // That is depending on if we want nested subtrees (layers) to bubble
+    // events to their parent. We could also go through parentNode on the
+    // host node but that wouldn't work for React Native and doesn't let us
+    // do the portal feature.
+  } while (fiber && fiber.tag !== HostComponent);
+  if (fiber) {
+    return fiber;
+  }
+  return null;
+}
+
+/**
+ * Find the deepest React component completely containing the root of the
+ * passed-in instance (for use when entire React trees are nested within each
+ * other). If React trees are not nested, returns null.
+ */
+function findRootContainerNode(inst) {
+  // TODO: It may be a good idea to cache this to prevent unnecessary DOM
+  // traversal, but caching is difficult to do correctly without using a
+  // mutation observer to listen for all DOM changes.
+  while (inst.return) {
+    inst = inst.return;
+  }
+  if (inst.tag !== HostRoot) {
+    // This can happen if we're in a detached tree.
+    return null;
+  }
+  return inst.stateNode.containerInfo;
+}
+
+function triggerEventHandler(
+  domNode,
+  domNodeEventsMap,
+  eventName,
+  proxyContext,
+  event,
+) {
+  const eventListener = domNodeEventsMap.get(eventName);
+  if (eventListener === undefined || eventListener.handler === null) {
+    return;
+  }
+  proxyContext.currentTarget = domNode;
+  invokeGuardedCallbackAndCatchFirstError(
+    eventName,
+    eventListener.handler,
+    undefined,
+    event,
+  );
+}
+
+function dispatchEventHandler(fiber, capturePhase, proxyContext) {
+  const domNode = getDOMNodeFromFiber(fiber);
+  // Html Nodes can be nested fe: span inside button in that scenario
+  // browser does not handle disabled attribute on parent,
+  // because the event listener is on document.body
+  // Don't process clicks on disabled elements
+  if (proxyContext.isClickEvent && domNode.disabled) {
```

Comment:
isn't this true for any mouse/pointer event?

Reply:
This was taken from Inferno and I believe we have something similar in the current ReactDOM inside SimpleEventsPlugin.